src/bindings/python/PyGpuShaderDesc.cpp

```diff
@@ -1,186 +1,337 @@
+            {
+                py::gil_scoped_release release;
 
-} // anon namespace
+                const char * textureName;
+                const char * samplerName;
+                const char * uid;
+                unsigned width, height;
+                GpuShaderDesc::TextureType channel;
+                Interpolation interpolation;
+                self->getTexture(index, textureName, samplerName, uid, width, height, channel, 
+                                 interpolation);
+
+                ssize_t numChannels;
+
+                switch (channel)
+                {
+                    case GpuShaderDesc::TEXTURE_RED_CHANNEL:
+                        numChannels = 1;
+                        break;
+                    case GpuShaderDesc::TEXTURE_RGB_CHANNEL:
+                        numChannels = 3;
+                        break;
+                    default:
+                        throw Exception("Error: Unsupported texture type");
+                }
+
+                const float * values;
+                self->getTextureValues(index, values);
+
+                py::gil_scoped_acquire acquire;
+
+                return py::array(py::dtype("float32"), 
+                                 { height * width * numChannels },
+                                 { sizeof(float) },
+                                 values);
+            },
+             "index"_a)
+
+        // 3D lut related methods
+        .def("getNum3DTextures", &GpuShaderDesc::getNum3DTextures)
+        .def("add3DTexture", [](GpuShaderDescRcPtr & self,
+                                const std::string & textureName, 
+                                const std::string & samplerName, 
+                                const std::string & uid,
+                                unsigned edgelen,
+                                Interpolation interpolation,
+                                const py::buffer & values)
+            {
+                py::buffer_info info = values.request();
+                checkBufferType(info, py::dtype("float32"));
+                checkBufferSize(info, edgelen*edgelen*edgelen*3);
+
+                py::gil_scoped_release release;
+
+                self->add3DTexture(textureName.c_str(), 
+                                   samplerName.c_str(), 
+                                   uid.c_str(),
+                                   edgelen,
+                                   interpolation,
+                                   static_cast<float *>(info.ptr));
+            },
+             "textureName"_a, "samplerName"_a, "uid"_a, "edgelen"_a, "interpolation"_a, "values"_a)
+        .def("get3DTextures", [](GpuShaderDescRcPtr & self) 
+            {
+                return Texture3DIterator(self);
+            })
+        .def("get3DTextureValues", [](GpuShaderDescRcPtr & self, unsigned index) 
+            {
+                py::gil_scoped_release release;
+
+                const char * textureName;
```

Comment:
Always assign a default value to variables i.e. the debug mode assigns 0, but the release mode does nothing.

Reply:
I've assigned these to `nullptr`